File: competitive_programming/codeforces/11_golang/1a_theatre_square.go
package main

import "fmt"

func theatreSquare(n, m, a int64) int64 {
	return ((n + a - 1) / a) * ((m + a - 1) / a)
}

func main() {
	if theatreSquare(6, 6, 4) != 4 || theatreSquare(1, 1, 1) != 1 {
		panic("theatre square failed")
	}
	fmt.Println("1A theatre square ok")
}

File: competitive_programming/codeforces/09_rust/1a_theatre_square.rs
fn theatre_square(n: i64, m: i64, a: i64) -> i64 {
    ((n + a - 1) / a) * ((m + a - 1) / a)
}

fn main() {
    assert_eq!(theatre_square(6, 6, 4), 4);
    assert_eq!(theatre_square(1, 1, 1), 1);
    println!("1A theatre square ok");
}
